File: data/java/tests/test_jsonschema_validate.java
package com.example;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class test_jsonschema_validate {
    private function_jsonschema_validate validator;

    @BeforeEach
    void setUp() {
        validator = new function_jsonschema_validate();
    }

    @Test
    void test_case_1() {
        String data = "{\"name\": \"John\", \"age\": 30}";
        String schema = "{\"type\": \"object\", \"properties\": {\"name\": {\"type\": \"string\"}, \"age\": {\"type\": \"integer\"}}, \"required\": [\"name\", \"age\"]}";
        assertEquals("True", validator.validate_json(data, schema));
    }

    @Test
    void test_case_2() {
        String data = "{\"name\": \"John\"}";
        String schema = "{\"type\": \"object\", \"properties\": {\"name\": {\"type\": \"string\"}, \"age\": {\"type\": \"integer\"}}, \"required\": [\"name\", \"age\"]}";
        assertEquals("False", validator.validate_json(data, schema));
    }

    @Test
    void test_case_3() {
        String data = "{\"name\": \"John\", \"age\": \"30\"}";
        String schema = "{\"type\": \"object\", \"properties\": {\"name\": {\"type\": \"string\"}, \"age\": {\"type\": \"integer\"}}, \"required\": [\"name\", \"age\"]}";
        assertEquals("False", validator.validate_json(data, schema));
    }

    @Test
    void test_case_4() {
        String data = "{}";
        String schema = "{\"type\": \"object\", \"properties\": {\"name\": {\"type\": \"string\"}, \"age\": {\"type\": \"integer\"}}, \"required\": [\"name\", \"age\"]}";
        assertEquals("False", validator.validate_json(data, schema));
    }

    @Test
    void test_case_5() {
        String data = "{\"name\": \"John\", \"age\": 30}";
        String schema = "{\"type\": \"object\", \"properties\": {\"name\": {\"type\": \"string\"}}, \"required\": [\"name\"]}";
        assertEquals("True", validator.validate_json(data, schema));
    }
}

File: code/test_suites/python/python/test_jsonschema_validate.py
import unittest
from function_jsonschema_validate import JSONValidator

class TestJSONValidator(unittest.TestCase):
    
    def setUp(self):
        self.validator = JSONValidator()

    def test_case_1(self):
        data = '{"name": "John", "age": 30}'
        schema = '{"type": "object", "properties": {"name": {"type": "string"}, "age": {"type": "integer"}}, "required": ["name", "age"]}'
        self.assertEqual(self.validator.validate_json(data, schema), "True")

    def test_case_2(self):
        data = '{"name": "John"}'
        schema = '{"type": "object", "properties": {"name": {"type": "string"}, "age": {"type": "integer"}}, "required": ["name", "age"]}'
        self.assertEqual(self.validator.validate_json(data, schema), "False")

    def test_case_3(self):
        data = '{"name": "John", "age": "30"}'
        schema = '{"type": "object", "properties": {"name": {"type": "string"}, "age": {"type": "integer"}}, "required": ["name", "age"]}'
        self.assertEqual(self.validator.validate_json(data, schema), "False")

    def test_case_4(self):
        data = '{}'
        schema = '{"type": "object", "properties": {"name": {"type": "string"}, "age": {"type": "integer"}}, "required": ["name", "age"]}'
        self.assertEqual(self.validator.validate_json(data, schema), "False")

    def test_case_5(self):
        data = '{"name": "John", "age": 30}'
        schema = '{"type": "object", "properties": {"name": {"type": "string"}}, "required": ["name"]}'
        self.assertEqual(self.validator.validate_json(data, schema), "True")

if __name__ == '__main__':
    unittest.main()
